File: bl-central-server/bl-sensor-data-collector/src/main/java/org/jesperancinha/logistics/sensor/collector/converter/BridgeConverter.java
package org.jesperancinha.logistics.sensor.collector.converter;

import org.jesperancinha.logistics.jpa.dao.Bridge;
import org.jesperancinha.logistics.jpa.dao.BridgeLog;
import org.jesperancinha.logistics.sensor.collector.data.BridgeLogDto;

public class BridgeConverter {
    public static BridgeLog toModel(BridgeLogDto bridgeLogDto, Bridge bridge) {
        return BridgeLog.builder()
            .checkInOut(bridgeLogDto.type())
            .timestamp(bridgeLogDto.timestamp())
            .bridge(bridge)
            .build();
    }
}

File: bl-central-server/bl-sensor-data-collector/src/main/kotlin/org/jesperancinha/logistics/sensor/collector/converter/BridgeConverter.kt
package org.jesperancinha.logistics.sensor.collector.converter

import org.jesperancinha.logistics.jpa.dao.*
import org.jesperancinha.logistics.sensor.collector.data.BridgeLogDto

object BridgeConverter {
    fun toModel(bridgeLogDto: BridgeLogDto, bridge: Bridge?): BridgeLog {
        return BridgeLog(
            checkInOut = bridgeLogDto.type,
            timestamp = bridgeLogDto.timestamp,
            bridge = bridge
        )
    }
}